sourced/ml/algorithms/id_splitter/features.py

```diff
@@ -0,0 +1,108 @@
+import logging
+import string
+import tarfile
+import warnings
+
+import numpy as np
+try:
+    from keras.preprocessing.sequence import pad_sequences
+except ImportError as e:
+    warnings.warn("Tensorflow or/and Keras are not installed, dependent functionality is "
+                  "unavailable.")
+
+
+MAXLEN = 40  # max length of sequence
+PADDING = "post"  # add padding values after input
+
+# CSV default parameters
+TOKEN_COL = 3
+TOKEN_SPLIT_COL = 4
+
+
+def read_identifiers(csv_loc, use_header=True, mode="r", maxlen=MAXLEN, token_col=TOKEN_COL,
+                     token_split_col=TOKEN_SPLIT_COL, shuffle=True):
+    """
+    Read and filter too long identifiers from CSV file.
+    :param csv_loc: location of CSV file
```

Comment:
You say CSV file but it's actually tared CSV file, think you should either update dovs/log or check path to see whether it is in fact a tar file or csv file that was inputted

Reply:
yes, this part should be updated according to @zurk comments because he releases dataset.